packages/astro/src/core/build/generate.ts

```diff
@@ -196,9 +196,12 @@ export async function generatePages(opts: StaticBuildOptions, internals: BuildIn
 		}
 	}
 
-	logger.info(null, `\n${bgGreen(black(` generating optimized images `))}`);
-	for (const imageData of getStaticImageList()) {
-		await generateImage(pipeline, imageData[1].options, imageData[1].path);
+	const staticImageList = Array.from(getStaticImageList())
+
+	if (staticImageList.length) logger.info(null, `\n${bgGreen(black(` generating optimized images `))}`); let count = 0;
+	for (const imageData of staticImageList) {
```

Comment:
You can get the index directly in the `for of` here, no need for a separate variable

Reply:
You can? I am not aware, and haven't found, a way to get the index from a `for... of`.

I could call the `.entries()` method and destructive the array I get, but I get a separate variable reassigned each time.

Reply:
Oh no, you're right. I forgot how Map works, derp

Reply:
Yeah, it's kind of annoying, but that's how it is. I'm gonna see what's the best way forward, changing the structure.